client/my-sites/themes/theme-showcase.jsx

```diff
@@ -337,6 +346,36 @@ class ThemeShowcase extends Component {
 		this.scrollToSearchInput();
 	};
 
+	getCollectionHeader = () => {
```

Comment:
This looks like a component to me. Maybe we should extract it from here in a separate file?

Reply:
We would need to prop drill many things. In the future, it would be nice to extract most of the things in the theme showcase component to a React Context and then create some components. At the time being it would just be a mess to create a component (we would need to pass props and functions from the ThemeShowcase to the new component)

Reply:
We'll only need to pass `tier` (string) and `onClick` (callback) props; we don't have that many props to pass to the component.

Reply:
Since in the ThemeShowcase component, we already have some functions that return JSX, and the header is only used within the ThemeShowcase component I see no need to extract it to a component (besides what I've already mentioned in my previous comment).

Reply:
I still think that extracting this in a separate component is better, but if you have strong feelings about it, let's leave it as it is.

> It would be nice to have it in a separate component but first, we would need to refactor the actual ThemeShowcase component

We don't need to do that; refactoring should be done incrementally. And passing two props wouldn't add much overhead. In the end, it's just two props.

> Since in the ThemeShowcase component, we already have some functions that return JSX, and the header is only used within the ThemeShowcase component (so it's not a reusable component) I see no need to extract it to a component (besides what I've already mentioned in my previous comment).

The component already has almost 600 LoC; personally, I find it hard to follow the code. Adding more code won't help with readability.